### ISSUE TITLE
False positive for E1111 assignment-from-no-return with numpy.matrix.getT()

### ISSUE THREAD

**Issue by Choc13** [PyCQA/pylint#3544]
### Steps to reproduce
1. Install pylint 2.5.0
2. Write the following code:
```
matrix = np.asmatrix([1])
transpose = matrix.getT()
```
3. Run `pylint`

### Current behavior
pylint outputs the following error:  

```
E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```

### Expected behavior
AFAIK [`numpy.matrix.getT`](https://numpy.org/doc/stable/reference/generated/numpy.matrix.getT.html?highlight=gett#numpy.matrix.getT) does in fact return a value, which is the transposed matrix. So I believe this is a false positive.

Interestingly, in another place I have some code which does division on the matrix before taking the transpose like this which doesn't report the error:

```
def foo(scalar: float) -> float:
    matrix = np.asmatrix([1]) / scalar
    transpose = matrix.getT()
```

But then if I replace `scalar` with a literal value such as `1.0` like so:

```
matrix = np.asmatrix([1]) / 1.0
transpose = matrix.getT()
```

pylint instead reports:

```
E1101: Instance of 'ndarray' has no 'getT' member (no-member)
```

### pylint --version output
2.5.0

**Comment by hippo91** [comment]
@Choc13 thanks for the report.
In fact `astroid`, which is the inference engine used by `pylint` cannot analyse some `numpy` sources as they are written in C.
In this situtation, `astroid` uses some *brains* which are pieces of code that mock the `numpy` classes. 
Supporting the `numpy.matrix` class represents a high amount of work. In the `numpy` documentation it is written:
*It is no longer recommended to use this class, even for linear algebra. Instead use regular arrays. The class may be removed in the future.*
I don't think it is wise to spend time to support something that will be removed in the future, that's why i close this issue.
